packages/components/src/components/Icon/isHeroIcon.ts

```diff
@@ -0,0 +1,8 @@
+import * as HeroOutlineIcons from "@heroicons/react/24/outline";
+import { IconName } from "./IconName";
+
+export const isHeroIcon = (
+  name: IconName
+): name is keyof typeof HeroOutlineIcons => {
```

Comment:
Isn't this just returning a boolean? Could you help me understand `name is keyof typeof HeroOutlineIcons`?

Reply:
I believe this is a safe(r) method to check if an item is present as key of an object? https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates